javatests/arcs/android/storage/service/StorageServiceEndToEndTest.kt

```diff
@@ -62,7 +63,11 @@ class StorageServiceEndToEndTest {
   val log = LogRule()
 
   private suspend fun buildManager() =
```

Comment:
You can change this to `CoroutineScope.buildManager()`, then just pass `this` as the scope.

Reply:
This only works if buildManager() is inside a runBlocking{} block. But we also have in suspend functions.

Reply:
We should pass the scope through those methods so it can be used to call buildManager.

Reply:
Or, maybe just create the manager and that pass that to the helper method as a parameter?

Reply:
If we make this change, I think it may also be possible to switch to using runBlockingTest instead of runBlocking.

Reply:
OK that works. I also tried runBlockingTest but got various java.lang.IllegalStateException: This job has not completed yet errors.

Reply:
Ok... we should try to fix that if we can, it's usually a sign that something is not structured well with our coroutines usage. It means, in general, that some coroutine has been launched and not shut down, so it's potentially something we are leaking.

Unfortunately, I've found it hard to track these issues down, the error message is not very helpful. Usually I end up doing "binary search with println" to zero in on the last suspension to occur before the test fails with the error. 

At a glance, everything seems to be OK with the setup here, let me stare at it a little more...

Reply:
Oh, a good potential culprit here is scheduler running on Dispatchers.Default. One thing to try: If you change the instantiation to SimpleSchedulerProvider(TestCoroutineScope().coroutineContext) does it fix the issue?

Reply:
Tried but the test still fails.

Reply:
We should try to figure out why.  It could be a coroutine, or a job/deferred that isn't being completed/cancelled.